obspy/core/tests/test_trace.py

```diff
@@ -2591,6 +2591,30 @@ def test_dtype_is_not_unnecessarily_changed(self):
             tr_float64.copy().interpolate(
                 1.0, method="lanczos", a=2).data.dtype, np.float64)
 
+    def test_set_trace_id(self):
+        """
+        Test setter of `id` property.
+        """
+        tr = Trace()
+        tr.stats.location = "00"
+        tr.id = "GR.FUR..HHZ"
+        self.assertEqual(tr.stats.network, "GR")
+        self.assertEqual(tr.stats.station, "FUR")
+        self.assertEqual(tr.stats.location, "")
+        self.assertEqual(tr.stats.channel, "HHZ")
+
+        # check that invalid ids will raise
+        def set_id_test(x):
+            tr.id = x
+
+        invalid_types = (True, False, -10, 0, 1.0, [1, 4, 3, 2], np.ones(4))
+        for id_ in invalid_types:
+            self.assertRaises(TypeError, set_id_test, id_)
+        invalid_ids = ("ABCD", "AB.CD", "AB.CD.00", "..EE", "....",
+                       "GR.FUR..HHZ.", "GR.FUR..HHZ...")
+        for id_ in invalid_ids:
+            self.assertRaises(ValueError, set_id_test, id_)
```

Comment:
You could also do

```python
with self.assertRaises(ValueError):
    tr.id = id_
```

Then the test is more explicit and nicer to look at :-)